lib/mongoid/extensions/date.rb

```diff
@@ -60,7 +61,9 @@ def mongoize(object)
               else
                 time = object.__mongoize_time__
               end
-              ::Time.utc(time.year, time.month, time.day)
+              if time.acts_like?(:time)
```

Comment:
For `Date`, we should check if the driver can handle `Date` type, and `mongoize` can just return Date.

Reply:
`__mongoize_time__` only returns a Time or `ActiveSupport::TimeWithZone`. I have added a test to demonstrate this.